src/main/kotlin/org/jdc/template/ux/startup/StartupViewModel.kt

```diff
@@ -42,6 +57,82 @@ class StartupViewModel
         }
     }
 
+    /**
+     * Creates Individual sample data WITH using injection
+     */
+    fun createSampleIndividualDataWithInjection() = launch(cc.commonPool) {
+        val call = individualService.individuals()
+
+        call.enqueue(object : Callback<DtoIndividuals> {
+            override fun onResponse(call: Call<DtoIndividuals>, response: Response<DtoIndividuals>) {
+                processWebIndividualServiceResponse(response)
+            }
+
+            override fun onFailure(call: Call<DtoIndividuals>, t: Throwable?) {
+                Timber.e(t, "Individual Search FAILED")
+            }
+        })
+    }
+
+    private fun processWebIndividualServiceResponse(response: Response<DtoIndividuals>) {
+        if (response.isSuccessful) {
+            Timber.i("Individual Search SUCCESS")
+            response.body()?.let {
+                processIndividualSearchResponse(it)
+            }
+        } else {
+            Timber.e("Search FAILURE: code (%d)", response.code())
+        }
+    }
+
+    private fun processIndividualSearchResponse(dtoIndividuals: DtoIndividuals) = launch(cc.commonPool) {
+        val list : MutableList<Individual> = mutableListOf()
+        val result : List<Individual> = list
+
+        val household = Household()
+        household.name = "Shan"
+
+        for (ind in dtoIndividuals.individuals) {
+
+            val thisIndividual = Individual()
+            thisIndividual.id = ind.id
+            thisIndividual.firstName = ind.firstName
+            thisIndividual.lastName = ind.lastName
+            thisIndividual.phone = "801-555-0000"
+            thisIndividual.individualType = IndividualType.HEAD
+            thisIndividual.householdId = household.id
+            val parts = ind.birthdate.split("-")
+            thisIndividual.birthDate = LocalDate.of(parts[0].toInt(), parts[1].toInt(), parts[2].toInt())
+            thisIndividual.alarmTime = LocalTime.of(7, 0)
+            thisIndividual.profileUrl = ind.profilePicture
+            list.add(thisIndividual)
+            Timber.i("Result: %s - %s", thisIndividual.firstName, thisIndividual.profileUrl)
+        }
+        persistIndividuals(result, household)
+    }
+
+    private fun persistIndividuals(individuals : List<Individual>, household: Household) {
+        val individualDao = mainDatabase.individualDao()
+        val householdDao = mainDatabase.householdDao()
+
+        // clear any existing items
```

Comment:
You need a try catch finally around this, which I believe Jeff mentioned elsewhere.